docs/preview/03-Features/01-Azure/01-service-bus.md

```diff
@@ -0,0 +1,279 @@
+    // this job instance in a multi-instance deployment (default: generated GUID).
+    options.JobId = Guid.NewGuid().ToString();
+
+    // Indicate whether or not the default built-in JSON deserialization should ignore additional members 
+    // when deserializing the incoming message (default: AdditionalMemberHandling.Error).
+    options.Routing.Deserialization.AdditionalMembers = AdditionalMemberHandling.Ignore;
+});
+```
+
+### Message handler routing customization
+The following routing options are available when registering an Azure Service Bus message handler on a message pump.
+
+> 💡 Setting one or more of the routing options helps the message pump to better select the right *message handler* for the received message.
+
+```csharp
+services.AddServiceBus[Topic/Queue]MessagePump(...)
+        .WithServiceBusMessageHandler<OrderMessageHandler, Order>(..., options =>
+        {
+            // Adds a filter to the message handler registration:
+            // Only messages with 'Type=Order' property goes through this message handler.
+            options.AddMessageContextFilter(ctx => ctx.Properties["Type"] == "Order");
+
+            // Adds a filter to the message handler registration:
+            // Only messages with certain bodies goes through this message handler.
+            options.AddMessageBodyFilter((Order order) => order.OrderId = "123");
+
+            // Adds a custom message deserializer to the message handler registration:
+            // Only messages that gets deserialized successfully goes through this message handler.
+            // 👉 Custom implementations of the `Arcus.Messaging.Abstractions.MessageHandling.IMessageBodySerializer` interface.
+            //      public interface IMessageBodySerializer
+            //      {
+            //          Task<MessageResult> DeserializeMessageAsync(string messageBody);
+            //      }
+            options.AddMessageBodySerializer(new CustomXmlMessageBodySerializer());
+        });
+```
+
+### Custom message settlement
+When messages can't be matched to any of your custom registered message handlers, Arcus will dead-letter the message. When one of your message handlers fails to process a message, it will get abandoned and maybe only after a third try be dead-lettered.
+
+This settlement of received Azure Service Bus messages can also be customized by calling one of the Service Bus operations yourself via the message context.
+
+```csharp
+public class OrderMessageHandler : IAzureServiceBusMessageHandler<Order>
+{
+    public async Task Task ProcessMessageAsync(
+        Order message,
+        AzureServiceBusMessageContext messageContext,
+        MessageCorrelationInfo correlation,
+        CancellationToken cancellation)
+    {
+        await messageContext.DeadLetterMessageAsync("Reason: Unsupported", "Message type is not supported");
+    }
+}
+```
+
+The following operations are supported:
+* **Dead-letter**
+* **Abandon**
+* **Complete**
```

Comment:
Also added the custom message settlement via the message context, added in PR #551